docs/developers_notes/basis_module.md

```diff
@@ -0,0 +1,130 @@
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be summed together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for an additive basis set.
+
+## The `MultiplicativeBasis` Class
+
+This class extends the `Basis` abstract class and represents a set of basis functions that are combined multiplicatively. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for multiplicative basis functions.
+
+### Overridden Methods
+
+1. `_evaluate`: For a `MultiplicativeBasis` object, this method should evaluate each basis function at the specified samples and then multiply the results together.
+2. `_get_samples`: This method should return equidistant samples over the domain of each basis function. The samples should then be multiplied together.
+3. `_check_n_basis_min`: This method checks whether the number of basis functions meets the minimum requirement for a multiplicative basis set.
+
+## The `SplineBasis` and `MSplineBasis` Classes
+
+The `SplineBasis` class is an abstract subclass of `Basis` that represents a spline basis function. The `MSplineBasis` class is a concrete subclass of `SplineBasis` that implements a specific type of spline basis, the M-spline.
+
+### Overridden Methods for `MSplineBasis`
+
+1. `_evaluate`: For an `MSplineBasis` object, this method should evaluate the M-spline at the specified samples.
+2. `_get_samples`: This method should return equidistant samples over the domain of the M-spline.
+3. `_check_n_basis_min`: This method checks whether the number of M-spline basis functions meets the minimum requirement.
+
+## The `RaisedCosineBasis`, `RaisedCosineBasisLinear`, and `RaisedCosineBasisLog` Classes
+
+The `RaisedCosineBasis` class is an abstract subclass of `Basis` that represents a raised cosine basis function. The `RaisedCosineBasisLinear` and `RaisedCosineBasisLog` classes are concrete subclasses of `RaisedCosineBasis` that implement specific types of raised cosine basis functions.
+
+### Overridden Methods for `RaisedCosineBasisLinear` and `RaisedCosineBasisLog`
+
+1. `_evaluate`: For these objects, this method should evaluate the raised cosine function (with either linear or logarithmic spacing) at the specified samples.
+2. `_get_samples`: This method should return equidistant samples over the domain of the raised cosine function.
+3. `_check_n_basis_min`: This method checks whether the number of raised cosine basis functions meets the minimum requirement.
+
+## The `OrthExponentialBasis` Class
+
+The `OrthExponentialBasis` class extends the `Basis` abstract class and represents an orthogonal exponential basis set. It overrides the `_evaluate`, `_get_samples`, and `_check_n_basis_min` methods with specific implementations for orthogonal exponential basis functions.
```

Comment:
needs a longer description or a link to what a "orthogonal exponential basis set" means.

Reply:
can you help me with this? 

I understand what is happening mechanistically with this basis, it is a basis of decaying exponential with different decaying rates that gets passed through a numerical orthonormalization (something like using Gram–Schmidt method). I am less sure of when could be useful and specific applications of it

Reply:
We should talk with @ahwillia about this, I believe he came up with this basis set.